### PR TITLE
fix(plugin-credentials): checks to disable refresh and getUserToken while loggingOut

### PR DESCRIPTION
**Pull request by arun3528**
Disable refresh and getUserToken while logging out

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -71,6 +71,18 @@ This is mostly useful for the the example app(s), but also comes in handy when d
 PACKAGE=PACKAGENAME npm run grunt:package -- serve
 ```
 
+### Running eslint on
```
Comment (ianwremmel) [inline README.md]: incomplete title?

File: README.md [hunk 2]
```diff
@@ -71,6 +71,18 @@ This is mostly useful for the the example app(s), but also comes in handy when d
...
+This is mostly useful for running the eslint on the packages
```
Comment (ianwremmel) [inline README.md]: Something more like "Run eslint in a single package"

File: README.md [hunk 3]
```diff
@@ -79,6 +91,10 @@ OK, this one's a handful and requires a global package, but there were too many
 npm install -g nodemon
 nodemon -w packages/PACKAGENAME/src -w packages/PACKAGENAME/test -x "UNIT_ONLY=true PACKAGE=PACKAGENAME npm run --silent grunt:package express:test test:node"
 ```
+Turning on verbose mode by setting env variable
```
Comment (ianwremmel) [inline README.md]: "Aggressively log network requests"

File: packages/plugin-credentials/test/unit/spec/credentials.js [hunk 4]
```diff
@@ -110,6 +110,13 @@ describe(`plugin-credentials`, () => {
         spark.credentials.supertoken = supertoken;
         spark.credentials.userTokens.add(apiToken);
         spark.credentials.userTokens.add(kmsToken);
+
+        spark.credentials.isLoggingOut = false;
```
Comment (ianwremmel) [inline packages/plugin-credentials/test/unit/spec/credentials.js]: Add "isLoggingOut" as a session property; then, you can default it to `false` and remove this line

File: packages/plugin-credentials/test/unit/spec/credentials.js [hunk 5]
```diff
@@ -110,6 +110,13 @@ describe(`plugin-credentials`, () => {
...
+      });
+
+      it(`not trigger refresh when logging out`, () => {
```
Comment (ianwremmel) [inline packages/plugin-credentials/test/unit/spec/credentials.js]:
```javascript
describe(`when logging out`, () => {
  it(`rejects`, () => ...)
});
```

File: packages/plugin-credentials/test/unit/spec/credentials.js [hunk 6]
```diff
@@ -282,6 +289,15 @@ describe(`plugin-credentials`, () => {
     });
 
     describe(`#refresh()`, () => {
+
+      beforeEach(() => {
```
Comment (ianwremmel) [inline packages/plugin-credentials/test/unit/spec/credentials.js]: Removable if isLoggingOut defaults to false

File: packages/plugin-credentials/test/unit/spec/credentials.js [hunk 7]
```diff
@@ -282,6 +289,15 @@ describe(`plugin-credentials`, () => {
...
+        spark.credentials.isLoggingOut = false;
+      });
+      it(`not trigger refresh when logging out`, () => {
```
Comment (ianwremmel) [inline packages/plugin-credentials/test/unit/spec/credentials.js]:
```javascript
describe(`when logging out`, () => {
  it(`rejects`, () => ...)
});
```

File: packages/plugin-conversation/src/conversation.js [hunk 8]
```diff
@@ -131,10 +131,11 @@ const Conversation = SparkPlugin.extend({
     const shunt = new EventEmitter();
     const promise = (isEncrypted ? this.spark.encryption.download(item.scr) : this._downloadUnencryptedFile(item.url))
       .on(`progress`, (...args) => shunt.emit(`progress`, ...args))
-      .then((res) => readExifData(item, res))
+      .then((res) => {
```
Comment (ianwremmel) [inline packages/plugin-conversation/src/conversation.js]: Why did you change this from an implicit return?

File: packages/plugin-conversation/src/conversation.js [hunk 9]
```diff
@@ -163,7 +164,9 @@ const Conversation = SparkPlugin.extend({
     };
 
     const promise = this.request(options)
-      .then((res) => res.body);
+      .then((res) => {
```
Comment (ianwremmel) [inline packages/plugin-conversation/src/conversation.js]: Why did you change this from an implicit return?

File: packages/plugin-credentials/src/credentials/common.js [hunk 10]
```diff
@@ -150,6 +154,10 @@ export default {
   getUserToken(scope) {
     // Note: this behaves much like oneFlight, but doesn't return a unique
     // promise. Since it recursively calls iteself, oneFlight is problematic.
+    if (this.isLoggingOut) {
+      return Promise.reject(`credentials: Cannot get UserToken while logging out`);
```
Comment (ianwremmel) [inline packages/plugin-credentials/src/credentials/common.js]: `credentials: Credentials#getUserToken() is not available while Credentials#isLoggingOut is true`

File: packages/plugin-credentials/src/credentials/common.js [hunk 11]
```diff
@@ -215,10 +224,12 @@ export default {
   @oneFlight
   @waitForValue(`@`)
   refresh() {
+    if (this.isLoggingOut) {
+      return Promise.reject(new Error(`credentials: Cannot refresh while logging out`));
```
Comment (ianwremmel) [inline packages/plugin-credentials/src/credentials/common.js]: `credentials: Credentials#refresh() is not available while Credentials#isLoggingOut is true`

File: packages/plugin-credentials/test/unit/spec/credentials.js [hunk 12]
```diff
@@ -110,6 +110,14 @@ describe(`plugin-credentials`, () => {
         spark.credentials.supertoken = supertoken;
         spark.credentials.userTokens.add(apiToken);
         spark.credentials.userTokens.add(kmsToken);
+
+      });
+
+      describe(`when logging out`, () => {
+        it(`rejects getUserToken`, () => {
```
Comment (ianwremmel) [inline packages/plugin-credentials/test/unit/spec/credentials.js]: `rejects`. no need to indicate getUserToken because you're already in its describe block.

File: packages/plugin-encryption/src/encryption.js [hunk 13]
```diff
@@ -34,7 +34,10 @@ const Encryption = SparkPlugin.extend({
 
         return scr.decrypt(b);
       })
-      .then(ensureBuffer);
+      .then((res) => {
```
Comment (ianwremmel) [inline packages/plugin-encryption/src/encryption.js]: This then block doesn't do anything.

File: packages/plugin-encryption/src/encryption.js [hunk 14]
```diff
@@ -34,7 +34,10 @@ const Encryption = SparkPlugin.extend({
...
+        return res;
+      })
+      // .then(ensureBuffer);
```
Comment (ianwremmel) [inline packages/plugin-encryption/src/encryption.js]: remove

**Comment by saurjai3** [comment]
closing as it is handled through a different PR.
